plugins/aap-backend/src/providers/AapResourceEntityProvider.ts

```diff
@@ -0,0 +1,170 @@
+import { PluginTaskScheduler, TaskRunner } from '@backstage/backend-tasks';
+import {
+  ANNOTATION_LOCATION,
+  ANNOTATION_ORIGIN_LOCATION,
+  Entity,
+  ResourceEntity,
+} from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+
+import { Logger } from 'winston';
+
+import { listJobTemplates } from '../clients/AapResourceConnector';
+import { JobTemplate } from '../clients/types';
+import { readAapApiEntityConfigs } from './config';
+import { AapConfig } from './types';
+
+export class AapResourceEntityProvider implements EntityProvider {
+  private readonly env: string;
+  private readonly baseUrl: string;
+  private readonly authorization: string;
+  private readonly owner: string;
+  private readonly system: string;
+  private readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(
+    configRoot: Config,
+    options: {
+      logger: Logger;
+      schedule?: TaskRunner;
+      scheduler?: PluginTaskScheduler;
+    },
+  ): AapResourceEntityProvider[] {
+    const providerConfigs = readAapApiEntityConfigs(configRoot);
+
+    if (!options.schedule && !options.scheduler) {
+      throw new Error('Either schedule or scheduler must be provided.');
+    }
+
+    return providerConfigs.map(providerConfig => {
+      if (!options.schedule && !providerConfig.schedule) {
+        throw new Error(
+          `No schedule provided neither via code nor config for AapResourceEntityProvider:${providerConfig.id}.`,
+        );
+      }
+
+      const taskRunner =
+        options.schedule ??
+        options.scheduler!.createScheduledTaskRunner(providerConfig.schedule!);
+
+      return new AapResourceEntityProvider(
+        providerConfig,
+        options.logger,
+        taskRunner,
+      );
+    });
+  }
+
+  private constructor(
+    config: AapConfig,
+    logger: Logger,
+    taskRunner: TaskRunner,
+  ) {
+    this.env = config.id;
+    this.baseUrl = config.baseUrl;
+    this.authorization = config.authorization;
+    this.owner = config.owner || 'AAP';
```

Comment:
Owner is a backstage entity reference. `AAP` is not a valid value.

https://backstage.io/docs/features/software-catalog/descriptor-format/#specowner-required-2